src/Polysemy/Internal/Tactics.hs

```diff
@@ -38,41 +40,36 @@ import Polysemy.Internal.Union
 -- data 'Polysemy.Resource.Resource' m a where
 --   'Polysemy.Resource.Bracket' :: m a -> (a -> m ()) -> (a -> m b) -> 'Polysemy.Resource.Resource' m b
 -- @
---
--- Here we have an @m a@ which clearly needs to be run first, and then
--- subsequently call the @a -> m ()@ and @a -> m b@ arguments. In a 'Tactical'
--- environment, we can write the threading code thusly:
---
+-- We can interpret this effect like so:
 -- @
--- 'Polysemy.Resource.Bracket' alloc dealloc use -> do
---   alloc'   <- 'runT'  alloc
---   dealloc' <- 'bindT' dealloc
---   use'     <- 'bindT' use
+-- runResource = Polysemy.interpretH $ \case
+--  'Polysemy.Resource.Bracket' alloc dealloc use -> do
+--    resource <- 'runTSimple' alloc
+--    result <- 'bindTSimple' use resource
+--    'bindTSimple' dealloc resource
+--    pure result
 -- @
---
--- where
---
+-- To run embedded higher-order actions (e.g. @alloc :: m a@s) we use 'runTSimple'.
+-- If we have a kleisli action (e.g. @use :: a -> m b@) we can instead use 'bindTSimple'.
+--
+-- There are however some cases where these functions are not sufficient.
+-- Consider the example of the 'Polysemy.Reader.local' action.
+-- In it, we want any inner computations to be executed with a __modified__ environment.
+-- This is not possible with 'runTSimple' and 'bindTSimple', because they automatically
+-- take the care to run all your embedded actions by using the current set of interpreters.
+-- As an example, let's look at the implementation for running a 'Polysemy.Reader.Reader'
+-- with a constant value:
 -- @
--- alloc'   ::         'Polysemy.Sem' ('Polysemy.Resource.Resource' ': r) (f a1)
--- dealloc' :: f a1 -> 'Polysemy.Sem' ('Polysemy.Resource.Resource' ': r) (f ())
--- use'     :: f a1 -> 'Polysemy.Sem' ('Polysemy.Resource.Resource' ': r) (f x)
+-- 'Polysemy.Reader.runReader' i = 'Polysemy.interpretH' $ \case
+--   'Polysemy.Reader.Ask' -> pureT i
+--   'Polysemy.Reader.Local' f m -> do
+--     mm <- 'runT' m
+--     'Polysemy.raise' $ 'Polysemy.Reader.runReader' (f i) mm
 -- @
---
--- The @f@ type here is existential and corresponds to "whatever
```

Comment:
Unfortunately there is no long an example to attach this talking point to, and it seems rather important, as it explains the that a `forall f. Functor f => f ...` is being used as "state".

Is there some other example that can be given here to demonstrate this? @KingoftheHomeless @tek

Reply:
can't we just put it at the end of the doc comment, so that the association with the type decl is apparent?

Reply:
ok obviously that should be "top of the comment" :smile: 
in any case, the first sentence mentions "threading the state", so it might be sensible to just expand on that there

Reply:
Added it back near the top + another piece of documentation that I previously deleted